### PR TITLE
#2 add tests

### PR DESCRIPTION
**Pull request by alexkart**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  | #2

### CODE REVIEW

File: src/PasswordHasher.php [hunk 1]
```diff
@@ -17,10 +18,10 @@ class PasswordHasher
 
     /**
      * @see https://www.php.net/manual/en/function.password-hash.php
-     * @param int|string $algorithm
+     * @param int $algorithm
      * @param array|null $parameters
      */
-    public function __construct($algorithm = PASSWORD_DEFAULT, array $parameters = null)
+    public function __construct(int $algorithm = PASSWORD_DEFAULT, array $parameters = null)
```
Comment (samdark) [inline src/PasswordHasher.php]: No, that should be left as is. See https://github.com/yiisoft/security/issues/5. That probably should be added to phpdoc.

**Comment by alexkart** [comment]
Is there a special reason why in one place `hash_hmac` prepended with `@` and in another place don't?
https://github.com/yiisoft/security/blob/master/src/Mac.php#L42
https://github.com/yiisoft/security/blob/master/src/Mac.php#L67
Or behaviors should be consistent?

File: tests/PasswordHasherTest.php [hunk 2]
```diff
@@ -27,4 +27,13 @@ public function testValidateEmptyPasswordException()
...
         $password->validate('', 'test');
     }
+
+    /**
+     * In PHP 7.4 password hashing algorithm identifiers are now nullable strings rather than integers.
+     */
+    public function testAlgorithmString()
+    {
+        $password = new PasswordHasher('test');
+        $this->assertTrue(true);
```
Comment (samdark) [inline tests/PasswordHasherTest.php]: Maybe it's better to assert that string is returned?

Comment (alexkart) [reply tests/PasswordHasherTest.php]: Returned from what?

**Comment by samdark** [comment]
It should be consistent regarding to `@` usage. I'd remove it and see how it goes.

**Comment by samdark** [comment]
Merged. Thank you!

File: tests/PasswordHasherTest.php (hunk 2 above)
Comment (samdark) [reply tests/PasswordHasherTest.php]: It's OK as is.

**Pull request merged**